app/src/features/Filters/FilterSelects.tsx

```diff
@@ -0,0 +1,161 @@
+    (state) => state.mapping.joins?.[filter.id ?? ""] ?? []
+  );
+  const filterColumn = filter.col ?? {};
+  const { table: PKTable, column: PKColumn } = filterColumn;
+  const isMappingPKTableAndFilterPKTableDifferent =
+    PKTable &&
+    mapping?.primary_key_table &&
+    PKTable !== mapping?.primary_key_table;
+
+  useEffect(() => {
+    if (joins.length === 0 && isMappingPKTableAndFilterPKTableDifferent) {
+      filter.id && dispatch(addJoin(filter.id));
+    }
+  }, [isMappingPKTableAndFilterPKTableDifferent]);
+
+  const handlePKTableChange = (table?: string) => {
+    onChange &&
+      onChange({
+        ...filter,
+        col: {
+          ...filterColumn,
+          table,
+        },
+      });
+  };
+  const handlePKColumnChange = (column?: string) => {
+    onChange &&
+      onChange({
+        ...filter,
+        col: {
+          ...filterColumn,
+          column,
+        },
+      });
+  };
+  const handleRelationChange = (
+    event: ChangeEvent<{ name?: string | undefined; value: unknown }>
+  ) => {
+    onChange &&
+      onChange({
+        ...filter,
+        relation: event.target.value as typeof filter.relation,
+      });
+  };
+  const handleValueChange = (
+    event: ChangeEvent<{ name?: string | undefined; value: unknown }>
+  ) => {
+    onChange &&
+      onChange({
+        ...filter,
+        value: event.target.value as string,
+      });
+  };
+  const handleFilterDelete = () => {
+    filter.id && dispatch(deleteFilter(filter.id));
+  };
+
+  return (
+    <Grid item container direction="column" spacing={2}>
+      <Grid item container xs={12} spacing={2}>
+        <ColumnSelects
+          owner={owner}
+          PKColumn={PKColumn}
+          PKTable={PKTable}
+          onPKTableChange={handlePKTableChange}
+          onPKColumnChange={handlePKColumnChange}
+        />
+        <Grid item>
+          <Select
+            value={filter.relation ?? ""}
+            options={FILTER_RELATIONS.map((relation) => ({
+              id: relation,
+              label: t(relation),
+            }))}
+            onChange={handleRelationChange}
+            emptyOption={t("selectOperation")}
+          />
+        </Grid>
+        <Grid item>
+          <TextField
+            className={classes.textInput}
+            value={filter.value}
+            onChange={handleValueChange}
+            placeholder={t("typeValue")}
+            variant="outlined"
+            size="small"
+          />
+        </Grid>
+        <Grid item>
+          <IconButton onClick={handleFilterDelete}>
+            <CloseIcon />
+          </IconButton>
+        </Grid>
+      </Grid>
+      {joins.length > 0 && (
+        <Grid item container>
+          <div className={classes.leftShift}>
+            <JoinSection
+              filter={filter}
+              owner={owner}
+              joins={joins}
+              isFirstJoinRequired={Boolean(
+                isMappingPKTableAndFilterPKTableDifferent
```

Comment:
it's already a boolean no?

Reply:
No, it could be `""`

Reply:
oh right, maybe cast `PKTable && mapping?.primary_key_table` as boolean then because the naming of this variable suggest it's a boolean